src/JLSOFile.jl

```diff
@@ -75,8 +75,9 @@ function JLSOFile(;
     image=_image(),
     kwargs...
 )
+    data = isempty(kwargs) ? Dict{Symbol,Any}() : Dict(kwargs)
     return JLSOFile(
-        Dict(kwargs);
+        data;
```

Comment:
i don't love this magic keyword  constructor, which has some "special" keywords (`version`, `julia`, ...) and then eats all the others as key/value pairs to be serialised... especially as we have a `:key => value` pair constructor already: `JLSOFile(data::Pair{Symbol}...; kwargs...) = JLSOFile(Dict(data); kwargs...)`

But I think removing this is another issue (and breaking). Maybe we should open an issue about that.

But whether we have magic-keywords or not, i think this is now the correct behjaviour for `JLSOFile()` (before it was a cryptic error)

Reply:
Yeah, :+1 for just using pairs for data objects. I think this was proposed cause some folks like creating dataframes with the keyword constructor and we often don't change the special keywords. Please open an issue to delete this method as it should be easy enough to deprecate.

Reply:
https://github.com/invenia/JLSO.jl/issues/109